azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/handlers/artifact/ArtifactHandlerImplV2.java

```diff
@@ -113,6 +124,13 @@ public void publish(final DeployTarget target) throws AzureExecutionException, I
         }
     }
 
+    protected void processResources() {
+        final Map<Boolean, List<Resource>> resourceMap = resources.stream()
+                .collect(Collectors.partitioningBy(resource -> isExternalResource(resource)));
+        resources = resourceMap.get(false);
```

Comment:
sometimes resources will be null and sometime externalResources will be null, you will get NPE afterwards

Reply:
Nice catch, fixed it by move it to deploy where will check the resources, and for externalResources, it will be empty list by `Collectors.partitioningBy` without NPE issue